src/Explorer/Tree/ResourceTreeAdapter.tsx

```diff
@@ -800,11 +800,7 @@ export class ResourceTreeAdapter implements ReactAdapter {
   }
 
   private cleanupDatabasesKoSubs(existingDatabaseIds: string[]): void {
-    const databaseIdsToRemove = this.databaseCollectionIdMap
-      .keys()
-      .filter((id: string) => existingDatabaseIds.indexOf(id) === -1);
-
-    databaseIdsToRemove.forEach((databaseId: string) => {
+    existingDatabaseIds.forEach((databaseId: string) => {
```

Comment:
I believe the previous code's purpose was to `dispose()` any existing subscription to a database which is not used anymore.
So you might want to consider using `this.databaseCollectionIdMap.keys()` (without the `filter()`'ing, which, as you pointed out, results in duplicates) rather than `existingDatabaseIds` to make sure it cleans up everything.

Reply:
Since we are lazy loading the collections now, the `this.databaseCollectionIdMap` will be empty before we expand the database node so we can't just use `this.databaseCollectionIdMap.keys()`.

However I do see your point that if a database is deleted, then it won't be in the `existingDatabaseIds` array so its subscription won't get cleaned up. I think the correct way is to use `this.koSubsDatabaseIdMap.keys()` in this case since it should contain all databases including the deleted ones.